shell/platform/fuchsia/flutter/flatland_platform_view.cc

```diff
@@ -136,6 +136,21 @@ void FlatlandPlatformView::OnChildViewStatus(
           });
 }
 
+void FlatlandPlatformView::OnChildViewViewRef(
+    uint64_t content_id,
+    uint64_t view_id,
+    fuchsia::ui::views::ViewRef view_ref) {
+  FML_DCHECK(child_view_info_.count(content_id) == 1);
```

Comment:
Should we FML_CHECK here instead?

You will get UB on like 147 if this precondition is not met.  I'd prefer a check over UB any day of the week

Reply:
Alternatively, keep this as a DCHECK and check that the result of at() is valid, then issue an FML_LOG(ERROR) if it is invalid

Reply:
This can absolutely happen if the dart code passes down the wrong view_id.  It has happened before with the GFX code and I had to go chasing it down in a P0 bug.

CHECK here means the dart code can DoS the runner.  It's a P0-inducing change.  I don't like P0's

Reply:
FlatlandPlatformView::OnChildViewViewRef isn't invoked by the dart code over a platform channel, its invoked from the callback passed to ChildViewWatcher::GetViewRef, which is called immediately after the map entry is created, which is why i never expect this check to fail (though I'm certainly not opposed to keeping the check in there). This also means that theres no error path to return an error to the dart code, and i was concerned that logging an error and returning might just cause the dart code to procede normally causing some other problem down the road (i.e. trying to set focus on it or something when the runner doesnt have the view ref because we dropped the view ref returning from this method early. I just figured if the state is borked things are going badly so its probably best to fall over and die as soon as we know something is wrong to make it easier to track down.

Reply:
Ohhh this is a FIDL callback from Flatland.  I completely missed that before.

Yeah, CHECK makes total sense here